requirements.txt

```diff
@@ -0,0 +1,222 @@
+#
+# This file is autogenerated by pip-compile with Python 3.11
+# by the following command:
+#
+#    pip-compile requirements.in
+#
+accelerate==1.1.1
+    # via
+    #   -r requirements.in
+    #   peft
+aiohappyeyeballs==2.4.3
+    # via aiohttp
+aiohttp==3.11.2
+    # via
+    #   datasets
+    #   fsspec
+aiosignal==1.3.1
+    # via aiohttp
+attrs==24.2.0
+    # via
+    #   aiohttp
+    #   jsonschema
+    #   referencing
+bitsandbytes==0.44.1
+    # via -r requirements.in
+certifi==2024.8.30
+    # via requests
+charset-normalizer==3.4.0
+    # via requests
+datasets==3.1.0
+    # via -r requirements.in
+dill==0.3.8
+    # via
+    #   datasets
+    #   multiprocess
+filelock==3.16.1
+    # via
+    #   datasets
+    #   huggingface-hub
+    #   torch
+    #   transformers
+    #   triton
+frozenlist==1.5.0
+    # via
+    #   aiohttp
+    #   aiosignal
+fsspec[http]==2024.9.0
+    # via
+    #   datasets
+    #   huggingface-hub
+    #   torch
+huggingface-hub==0.26.2
+    # via
+    #   accelerate
+    #   datasets
+    #   peft
+    #   tokenizers
+    #   transformers
+idna==3.10
+    # via
+    #   requests
+    #   yarl
+jinja2==3.1.4
+    # via torch
+jsonschema==4.23.0
+    # via valohai-yaml
+jsonschema-specifications==2024.10.1
+    # via jsonschema
+leval==1.3.0
+    # via valohai-yaml
+markupsafe==3.0.2
+    # via jinja2
+mpmath==1.3.0
+    # via sympy
+multidict==6.1.0
+    # via
+    #   aiohttp
+    #   yarl
+multiprocess==0.70.16
+    # via datasets
+networkx==3.4.2
+    # via torch
+numpy==2.1.3
+    # via
+    #   accelerate
+    #   bitsandbytes
+    #   datasets
+    #   pandas
+    #   peft
+    #   transformers
+nvidia-cublas-cu12==12.4.5.8
+    # via
+    #   nvidia-cudnn-cu12
+    #   nvidia-cusolver-cu12
+    #   torch
+nvidia-cuda-cupti-cu12==12.4.127
+    # via torch
+nvidia-cuda-nvrtc-cu12==12.4.127
+    # via torch
+nvidia-cuda-runtime-cu12==12.4.127
+    # via torch
+nvidia-cudnn-cu12==9.1.0.70
+    # via torch
+nvidia-cufft-cu12==11.2.1.3
+    # via torch
+nvidia-curand-cu12==10.3.5.147
+    # via torch
+nvidia-cusolver-cu12==11.6.1.9
+    # via torch
+nvidia-cusparse-cu12==12.3.1.170
+    # via
+    #   nvidia-cusolver-cu12
+    #   torch
+nvidia-nccl-cu12==2.21.5
+    # via torch
+nvidia-nvjitlink-cu12==12.4.127
+    # via
+    #   nvidia-cusolver-cu12
+    #   nvidia-cusparse-cu12
+    #   torch
+nvidia-nvtx-cu12==12.4.127
+    # via torch
```

Comment:
Mmh, these should probably not necessarily be here (you could do an `--index-url=https://download.pytorch.org/whl/cpu` while compiling so you get Torch wheels that are not Nvidia flavored).

Reply:
🤔

I do locally use a GPU (even throught the Docker image), so there kinda exists two sets of dependencies

could change to only use the CPU variant though, for simplicity

Reply:
would get rid of these bad boys:

```
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 28.00 MiB. GPU 0 has a total capacity of 7.79 GiB of which 26.50 MiB is free. Process 149428 has 3.96 GiB memory in use. Process 150711 has 3.75 GiB memory in use. Of the allocated memory 3.77 GiB is allocated by PyTorch, and 9.90 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation. See documentation for Memory Management
```

when I forget that I have something running already

:laughing:

Reply:
I tried running it with a CPU-only image:

```
RuntimeError: CUDA is required but not available for bitsandbytes. Please consider installing the multi-platform enabled version of bitsandbytes, which is currently a work in progress.
```

so, `bitsandbytes` that is used for the quantization is released as **GPU**-only?

I can go through with [installing the **alpha** multi-platform version](https://huggingface.co/docs/bitsandbytes/main/en/installation#multi-backend-supported-backends) (Intel CPU), but based on the `environment: trial2023-g4dn-xlarge` in the YAML file, this was intended to be showcased on GPU instances (`g4dn` being a T4 GPU instance type)

did I understand something wrong in the original comment? :upside_down_face:

Reply:
might, but even the base `torch` dependency is different on CPU vs. GPU `torch==2.5.1+cpu` :shrug: